doc/obabel.1
.Dd May 26, 2026
.Dt OBABEL 1 URM
.Os "Open Babel" 3.2
.Sh NAME
.Nm obabel
.Nd "a converter for chemistry and molecular modeling data files"
.Sh SYNOPSIS
.Nm
.Fl H Ns Ar help-options
.Nm
.Op Ar OPTIONS
.Op Fl i Ns Ar input-type | Fl : Ns Ar SMILES-string
.Ar infile
.Op Fl o Ns Ar output-type
.Fl O Ns Ar outfile
.Sh DESCRIPTION
.Nm
is a cross\-platform program designed to interconvert
between many file formats used in molecular modeling and computational
chemistry and related areas.
.Pp
Open Babel is also a complete programmers toolkit for developing
chemistry software.
For more information, see the Open Babel web pages
\%<\fBhttps://openbabel.org/\fR>.
.Sh OPTIONS
If only input and output files are given, Open Babel will guess the
file type from the filename extension.
.Bl -tag -width flag
.It Fl : Ns Ar SMILES-string
Enter SMILES string and use it in place of an input file.
The
.Ar SMILES-string
might need to be enclosed in quotation marks to avoid shell interpretation.
More than one can be used, and a molecule title can be included if enclosed in
quotes.
.It Fl a Ns Ar options
Format\-specific input options.
See
.Fl H Ns Ar format-ID
for options allowed by a particular format
.It Fl -addtotitle
Append text to the current molecule title
.It Fl -addformula
Append the molecular formula after the current molecule title
.It Fl b
Convert dative bonds: e.g., [N+]([O\-])=O to N(=O)=O
.It Fl c
Center atomic coordinates at (0,0,0)
.It Fl C
Combine molecules in first file with others having the same name
.It Fl e
Continue after errors
.It Fl d
Delete Hydrogens
.It Fl --errorlevel Ar n
Filter the level of errors and warnings displayed:
       1 = critical errors only
       2 = include warnings too (default)
       3 = include informational messages too
       4 = include \(lqaudit log\(rq messages of changes to data
       5 = include debugging messages too
.It Fl f Ar #
For multiple entry input, start import with molecule # as the first entry
.It Fl F
Output the available fingerprint types
.It Fl h
Add hydrogens
.It Fl H
Output usage information
.It Fl H Ns Ar format-ID
Output formatting information and options for format specified
.It Fl Hall
Output formatting information and options for all formats
.It Fl i Ns Ar format-ID
Specifies input format, see below for the available formats
.It Fl j , -join
Join all input molecules into a single output molecule entry
.It Fl k
Translate computational chemistry modeling keywords (e.g., GAMESS and Gaussian)
.It Fl m
Produce multiple output files, to allow:
.Bl -dash -offset indent -compact
.It
Splitting one input file \(em put each molecule into consecutively
numbered output files
.It
Batch conversion \(em convert each of multiple input files into a
specified output format
.El
.It Fl l Ar #
For multiple entry input, stop import with molecule # as the last entry
.It Fl o Ns Ar format-ID
Specifies output format, see below for the available formats
.It Fl O Ns Ar outfile
Specify the output file
.It Fl p
Add Hydrogens appropriate for pH (use transforms in phmodel.txt)
.It Fl -property
Add or replace a property (e.g., in an MDL SD file)
.It Fl s Ar SMARTS
Convert only molecules matching the SMARTS pattern specified
.It Fl -separate
Separate disconnected fragments into individual molecular records
.It Fl t
All input files describe a single molecule
.It Fl -title Ar title
Add or replace molecular title
.It Fl x Ns Ar options
Format\-specific output options.
See
.Fl H Ar format-ID
for options allowed by a particular format
.It Fl v Ar SMARTS
Convert only molecules \fBNOT\fP matching SMARTS pattern specified
.It Fl V
Output version number and exit
.It Fl z
Compress the output with gzip
.El
.Sh "FILE FORMATS"
The following formats are currently supported by Open Babel:
.Bl -item -offset indent -compact
.It
abinit \(em ABINIT Output Format [Read\-only]
.It
acesin \(em ACES input format [Write\-only]
.It
acesout \(em ACES output format [Read\-only]
.It
acr \(em ACR format [Read\-only]
.It
adf \(em ADF cartesian input format [Write\-only]
.It
adfband \(em ADF Band output format [Read\-only]
.It
adfdftb \(em ADF DFTB output format [Read\-only]
.It
adfout \(em ADF output format [Read\-only]
.It
alc \(em Alchemy format
.It
aoforce \(em Turbomole AOFORCE output format [Read\-only]
.It
arc \(em Accelrys/MSI Biosym/Insight II CAR format [Read\-only]
.It
ascii \(em ASCII format [Write\-only]
.It
axsf \(em XCrySDen Structure Format [Read\-only]
.It
bgf \(em MSI BGF format
.It
box \(em Dock 3.5 Box format
.It
bs \(em Ball and Stick format
.It
c09out \(em Crystal 09 output format [Read\-only]
.It
c3d1 \(em Chem3D Cartesian 1 format
.It
c3d2 \(em Chem3D Cartesian 2 format
.It
cac \(em CAChe MolStruct format [Write\-only]
.It
caccrt \(em Cacao Cartesian format
.It
cache \(em CAChe MolStruct format [Write\-only]
.It
cacint \(em Cacao Internal format [Write\-only]
.It
can \(em Canonical SMILES format
.It
car \(em Accelrys/MSI Biosym/Insight II CAR format [Read\-only]
.It
castep \(em CASTEP format [Read\-only]
.It
ccc \(em CCC format [Read\-only]
.It
cdjson \(em ChemDoodle JSON
.It
cdx \(em ChemDraw binary format [Read\-only]
.It
cdxml \(em ChemDraw CDXML format
.It
cht \(em Chemtool format [Write\-only]
.It
cif \(em Crystallographic Information File
.It
cjson \(em Chemical JSON
.It
ck \(em ChemKin format
.It
cml \(em Chemical Markup Language
.It
cmlr \(em CML Reaction format
.It
cof \(em Culgi object file format
.It
com \(em Gaussian Input [Write\-only]
.It
confabreport \(em Confab report format [Write\-only]
.It
CONFIG \(em DL\-POLY CONFIG
.It
CONTCAR \(em VASP format
.It
CONTFF \(em MDFF format
.It
copy \(em Copy raw text [Write\-only]
.It
crk2d \(em Chemical Resource Kit diagram(2D)
.It
crk3d \(em Chemical Resource Kit 3D format
.It
csr \(em Accelrys/MSI Quanta CSR format [Write\-only]
.It
cssr \(em CSD CSSR format [Write\-only]
.It
ct \(em ChemDraw Connection Table format
.It
cub \(em Gaussian cube format
.It
cube \(em Gaussian cube format
.It
dallog \(em DALTON output format [Read\-only]
.It
dalmol \(em DALTON input format
.It
dat \(em Generic Output file format [Read\-only]
.It
dmol \(em DMol3 coordinates format
.It
dx \(em OpenDX cube format for APBS
.It
ent \(em Protein Data Bank format
.It
exyz \(em Extended XYZ cartesian coordinates format
.It
fa \(em FASTA format
.It
fasta \(em FASTA format
.It
fch \(em Gaussian formatted checkpoint file format [Read\-only]
.It
fchk \(em Gaussian formatted checkpoint file format [Read\-only]
.It
fck \(em Gaussian formatted checkpoint file format [Read\-only]
.It
feat \(em Feature format
.It
fh \(em Fenske\-Hall Z\-Matrix format [Write\-only]
.It
fhiaims \(em FHIaims XYZ format
.It
fix \(em SMILES FIX format [Write\-only]
.It
fps \(em FPS text fingerprint format (Dalke) [Write\-only]
.It
fpt \(em Fingerprint format [Write\-only]
.It
fract \(em Free Form Fractional format
.It
fs \(em Fastsearch format
.It
fsa \(em FASTA format
.It
g03 \(em Gaussian Output [Read\-only]
.It
g09 \(em Gaussian Output [Read\-only]
.It
g16 \(em Gaussian Output [Read\-only]
.It
g92 \(em Gaussian Output [Read\-only]
.It
g94 \(em Gaussian Output [Read\-only]
.It
g98 \(em Gaussian Output [Read\-only]
.It
gal \(em Gaussian Output [Read\-only]
.It
gam \(em GAMESS Output [Read\-only]
.It
gamess \(em GAMESS Output [Read\-only]
.It
gamin \(em GAMESS Input
.It
gamout \(em GAMESS Output [Read\-only]
.It
gau \(em Gaussian Input [Write\-only]
.It
gjc \(em Gaussian Input [Write\-only]
.It
gjf \(em Gaussian Input [Write\-only]
.It
got \(em GULP format [Read\-only]
.It
gpr \(em Ghemical format
.It
gr96 \(em GROMOS96 format [Write\-only]
.It
gro \(em GRO format
.It
gukin \(em GAMESS\-UK Input
.It
gukout \(em GAMESS\-UK Output
.It
gzmat \(em Gaussian Z\-Matrix Input

NEWS
Open Babel 2.5.0 (2017-XX-XX)

Open Babel 2.4.0 (2016-9-15)

Open Babel 2.3.0 (2010-10-23)

This release represents a major update and should be a stable upgrade,
strongly recommended for all users of Open Babel. Highlights include a
completely rewritten stereochemistry engine, Spectrophore fingerprint
generation, 2D depiction, improved 3D coordinate generation, conformer
searching, and more. Many formats are improved or added, including
CIF, PDBQT, SVG, and more. Improved developer API and scripting
support and many, many bug fixes are also included.

What's New from 2.2.3:

* Completely rewritten stereochemistry perception, including support
  for tetrahedral, square planar, and higher-order stereochemistry.
* Dramatically improved canonicalization algorithm (Note that in general, canonical SMILES have changed since the 2.2.x release.)
* 2D depiction, including SVG vector graphics generation using code from MCDL.
* New Spectrophore generation, contributed by Silicos NV.
* New ChargeMethod API including support for partial charge assignment
  from Gasteiger, MMFF94, QEq, QTPIE methods and plugin interface for
  adding more.
* Improved 3D coordinate generation.
* New conformer generation framework, including support for diverse
  conformer generation and genetic algorithm lowest-energy searching.
* Improved user documentation.
* Improved aromaticity / Kekule bond assignment.
* Improved unit test suite using the CMake-based CTest program.
* Improved support for crystallographic unit cells (e.g., in CIF format).
* Improved UFF force field method, including hypervalent 5, 6, 7 and higher coordination numbers.
* Support for the GAFF (Generalized Amber Force Field) method.
* Support for reading geometry optimizations as multiple conformers
  from Gaussian, GAMESS-US, and other quantum chemistry packages.
* Support for reading molecular orbital energies from quantum chemistry formats.
* Several memory leaks fixed.
* Fixed many compiler warnings.
* Fixed support for MinGW and Cygwin environments.
* Fixed bugs with Gaussian 09 output files.
* Latest released version of the InChI library (1.0.3) generating standard InChI.
* Many more bug fixes and small feature improvements.

==New Command-Line Operations==

* --canonical: Output atoms in canonical order for any file format (i.e., not just SMILES)
* --conformer: Run a conformer search on the input molecules (has many options)
* --gen2D: Generate a 2D depiction of the molecule
* --partialcharge <model>: Use the partial charge model supplied to
  generate charges (i.e., instead of default Gasteiger sigma model)
* --sort <descriptor>: Sort molecules by a specified descriptor
* --unique: Only output unique molecules (as determined by InChI generation)

==New File Formats==

Import & Export:
- DL-POLY CONFIG
- FHIaims XYZ
- PDBQT

Import only:
- DL-POLY HISTORY
- GULP output
- PWscf output
- Text

Export only:
- MNA (Multilevel Neighborhoods of Atoms)
- SVG vector graphics


Open Babel 2.2.3 (2009-07-31)

What's New from 2.2.2:

This release represents an important bug-fix upgrade, strongly recommended
for all users of Open Babel.

- Fixed bug in fingerprints in 2.2.2, where the default fingerprint
  format and bit length was changed inadvertently.
- Fixed detection of shared_ptr in tr1/memory.
- Fixed additional aromaticity / Kekule assignment bugs.
- Fixed several bugs in the MMCIF format.
- Additional bug fixes.

Open Babel 2.2.2 (2009-07-04)

What's New from 2.2.1:

This release represents a major bug-fix release and is a stable
upgrade, strongly recommended for all users of Open Babel. While there
may not be many new features, many crashes and other bugs have been
fixed since 2.2.1.

- Upgraded to the new InChI 1.02 release to produce standardized InChI
  and InChIKey output.

tools/obabel.cpp
/**********************************************************************
main.cpp - Main conversion program, command-line handling.

Copyright (C) 1998-2001 by OpenEye Scientific Software, Inc.
Some portions Copyright (C) 2001-2006 by Geoffrey R. Hutchison
Some portions Copyright (C) 2004-2010 by Chris Morley

This file is part of the Open Babel project.
For more information, see <http://openbabel.org/>

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation version 2 of the License.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.
***********************************************************************/

// used to set import/export for Cygwin DLLs
#ifdef WIN32
#define USING_OBDLL
#endif

#include <openbabel/babelconfig.h>

#include <fstream>
#include <iostream>
#include <string>
#if HAVE_CONIO_H
#include <conio.h>
#endif

#if !HAVE_STRNCASECMP
extern "C" int strncasecmp(const char* s1, const char* s2, size_t n);
#endif

#include <openbabel/obconversion.h>
#include <openbabel/plugin.h>

using namespace std;
using namespace OpenBabel;

void DoOption(const char* p, OBConversion& Conv, OBConversion::Option_type typ,
              int& arg, int argc, char* argv[]);
void usage();
void help();

// There isn't a great way to do this -- we need to save argv[0] for usage()
namespace {
char* program_name;
}

int main(int argc, char* argv[]) {
  OBConversion Conv{&cin, &cout};  // default input and output are console

  OBFormat* pInFormat = nullptr;
  OBFormat* pOutFormat = nullptr;
  bool outGzip = false;
  vector<string> FileList, OutputFileList;
  string OutputFileName;

  // Parse commandline
  bool gotInType = false, gotOutType = false;
  bool SplitOrBatch = false;

  char* oext = nullptr;

  // Save name of program without its path (and .exe)
  string pn{argv[0]};
  string::size_type pos;
#ifdef _WIN32
  pos = pn.find(".exe");
  if (pos != string::npos) argv[0][pos] = '\0';
#endif
  pos = pn.find_last_of("/\\");
  if (pos == string::npos)
    program_name = argv[0];
  else
    program_name = argv[0] + pos + 1;

  const char* p;
  for (int arg = 1; arg < argc; ++arg) {
    if (!argv[arg]) continue;
    if (argv[arg][0] != '-') {
      FileList.push_back(argv[arg]);
      continue;
    }

    char opchar[2] = "?";
    opchar[0] = argv[arg][1];
    switch (opchar[0]) {
      case 'V': {
        cout << "Open Babel " << BABEL_VERSION << " -- " << __DATE__ << " -- "
             << __TIME__ << endl;
        return 0;
      }

      case 'i': {
        // Parameter is the input format which overrides any file extensions
        gotInType = true;
        char* iext = argv[arg] + 2;
        if (!*iext)
          iext = argv[++arg];  // space left after -i: use next argument

        if (iext && strncasecmp(iext, "MIME", 4) == 0) {
          // get the MIME type from the next argument
          iext = argv[++arg];
          pInFormat = Conv.FormatFromMIME(iext);
        } else {
          pInFormat = Conv.FindFormat(iext);
        }
        if (pInFormat == nullptr) {
          cerr << program_name << ": cannot read input format!" << endl;
          usage();
          return 1;
        }
        break;
      }

      case 'o':
        // Parameter is the output format which overrides any file extension
        gotOutType = true;
        oext = argv[arg] + 2;
        if (!*oext)
          oext = argv[++arg];  // space left after -i: use next argument

        if (oext && strncasecmp(oext, "MIME", 4) == 0) {
          // get the MIME type from the next argument
          oext = argv[++arg];
          pOutFormat = Conv.FormatFromMIME(oext);
        } else {
          pOutFormat = Conv.FindFormat(oext);
        }

        if (pOutFormat == nullptr) {
          cerr << program_name << ": cannot write output format!" << endl;
          usage();
          return 1;
        }
        break;

      case 'O':
        OutputFileName = argv[arg] + 2;
        if (OutputFileName.empty())
          OutputFileName =
              argv[++arg];  // space left after -O: use next argument
        break;

      case 'L': {  // display a list of plugin type or classes
        const char* param = nullptr;
        if (argc > arg + 1) param = argv[arg + 2];

        // First assume first arg is a plugin type and
        // param is a subtype, like babel -L ops gen3D
        // or first arg is a plugin ID, like babel -L cml
        OBPlugin* plugin;
        if ((OBPlugin::GetPlugin("plugins", argv[arg + 1]) &&
             (plugin = OBPlugin::GetPlugin(argv[arg + 1], param))) ||
            (plugin = OBPlugin::GetPlugin(nullptr, argv[arg + 1]))) {
          // Output details of subtype
          string txt;
          plugin->Display(txt, "verbose", argv[arg + 1]);
          cout << "One of the " << plugin->TypeID() << '\n' << txt << endl;
          return 0;
        }
        //...otherwise assume it is a plugin type, like babel -L forcefields
        // Output list of subtypes
        OBPlugin::List(argv[arg + 1], param);
        return 0;
      }

      case '?':
      case 'H':
        if (isalnum(argv[arg][2]) || arg == argc - 2) {
          if (strncasecmp(argv[arg] + 2, "all", 3)) {
            OBFormat* pFormat = (arg == argc - 2)
                                    ? Conv.FindFormat(argv[arg + 1])
                                    : Conv.FindFormat(argv[arg] + 2);
            if (pFormat) {
              cout << argv[arg] + 2 << "  " << pFormat->Description() << endl;
              if (pFormat->Flags() & NOTWRITABLE)
                cout << " This format is Read-only" << endl;
              if (pFormat->Flags() & NOTREADABLE)
                cout << " This format is Write-only" << endl;

              if (strlen(pFormat->SpecificationURL()))
                cout << "Specification at: " << pFormat->SpecificationURL()
                     << endl;
            } else {
              cout << "Format type: " << argv[arg] + 2 << " was not recognized"
                   << endl;
            }
          } else {
            OBPlugin::List("formats", "verbose");
          }
        } else {
          help();
        }
        return 0;

      case '-': {  // long option --name text
        // Option's text is in the next and subsequent args, until one starts
        // with -
        char* nam = argv[arg] + 2;
        if (!strcasecmp(nam, "help"))  // special case handled here
        {
          help();
          return 0;
        }
        if (*nam != '\0')  // Do nothing if name is empty
        {
          string txt;
          while (arg < argc - 1 && *argv[arg + 1] != '-') {
            // use text from subsequent args
            if (!txt.empty()) txt += ' ';  //..space separated if more than one
            txt += argv[++arg];
          }

          // If a API directive, e.g.---errorlevel
          // send to the pseudoformat "obapi" (without any leading -)
          if (*nam == '-') {
            OBConversion apiConv;
            OBFormat* pAPI = OBConversion::FindFormat("obapi");
            if (pAPI) {
              apiConv.SetOutFormat(pAPI);
              apiConv.AddOption(nam + 1, OBConversion::GENOPTIONS, txt.c_str());
              apiConv.Write(nullptr, &std::cout);
            }
          } else {
            // Is a normal long option name, e.g --addtotitle
            Conv.AddOption(nam, OBConversion::GENOPTIONS, txt.c_str());
          }
        }
      } break;

      case 'm':  // multiple output files
        SplitOrBatch = true;
        break;

      case 'a':  // single character input option
        p = argv[arg] + 2;
        DoOption(p, Conv, OBConversion::INOPTIONS, arg, argc, argv);
        break;

      case 'x':  // single character output option
        p = argv[arg] + 2;
        DoOption(p, Conv, OBConversion::OUTOPTIONS, arg, argc, argv);
        break;

      // Not essential, but allows these options to be before input filenames
      // since we know they take one parameter, and are the most likely options
      // to be misplaced
      case 'f':
      case 'l':
        p = argv[arg] + 2;
        if (!*p) p = argv[++arg];  // space left after -f: use next argument
        Conv.AddOption(opchar, OBConversion::GENOPTIONS, p);
        break;

      case ':':
        // e.g. -:c1ccccc1. SMILES passed as a file name and handled in
        // OBConversion
        FileList.push_back(argv[arg]);
        break;

      default:  // single character general option
        p = argv[arg] + 1;
        DoOption(p, Conv, OBConversion::GENOPTIONS, arg, argc, argv);
        break;
    }
  }

#if defined(_WIN32) && defined(USING_DYNAMIC_LIBS)
  // Expand wildcards in input filenames and add to FileList
  vector<string> tempFileList{FileList};
  FileList.clear();
  for (const auto& f : tempFileList) {
    if (f[0] == '-')
      FileList.push_back(f);
    else
      DLHandler::findFiles(FileList, f);
  }
#endif

  if (!gotInType && FileList.empty()) {
    cerr << "No input file or format spec or possibly a misplaced option.\n"
            "Most options must come after the input files. (-i -o -O -m can be "
            "anywhwere.)\n"
         << endl;
    usage();
    return 1;
  }

  if (!gotOutType) {
    // check there is a valid output format, but the extension will be
    // re-interpreted in OBConversion
    pOutFormat = Conv.FormatFromExt(OutputFileName.c_str(), outGzip);
    if (OutputFileName.empty() || pOutFormat == nullptr) {
      cerr << "Missing or unknown output file or format spec or possibly a "
              "misplaced option.\n"
              "Options, other than -i -o -O -m, must come after the input "
              "files.\n"
           << endl;
      usage();
      return 1;
    }
  }

  // rely on autodetection for gzipped input
  if (!Conv.SetInFormat(pInFormat)) {
    cerr << "Invalid input format" << endl;
    usage();
    return 1;
  }
  if (!Conv.SetOutFormat(pOutFormat, outGzip)) {
    cerr << "Invalid output format" << endl;
    usage();
    return 1;
  }

  if (SplitOrBatch) {
    // Put * into output file name before extension (or ext.gz)
    if (OutputFileName.empty()) {
      OutputFileName = "*.";
      OutputFileName += oext;
    } else {
      string::size_type pos = OutputFileName.rfind(".gz");
      if (pos == string::npos)
        pos = OutputFileName.rfind('.');
      else
        pos = OutputFileName.rfind('.', pos - 1);
      if (pos == string::npos)
        OutputFileName += '*';
      else
        OutputFileName.insert(pos, "*");
    }
  }

  int count = Conv.FullConvert(FileList, OutputFileName, OutputFileList);

  Conv.ReportNumberConverted(count);

  if (OutputFileList.size() > 1) {
    clog << OutputFileList.size() << " files output. The first is "
         << OutputFileList[0] << endl;
  }

  // std::string messageSummary = obErrorLog.GetMessageSummary();
  // if (messageSummary.size())
  //  {
  //    clog << messageSummary << endl;
  //  }

#ifdef _DEBUG
  // CM keep window open
  cout << "Press any key to finish" << endl;
  getch();
#endif

  return 0;
}

void DoOption(const char* p, OBConversion& Conv, OBConversion::Option_type typ,
              int& arg, int argc, char* argv[]) {
  // Unlike babel, cannot have multiple concatenated single char options
  // accepts: -sCCC -s CCC -s"CCC" -s CCC red -sCCC red
  char ch[2] = "?";
  *ch = *p++;
  std::string txt;
  // Get the option text
  if (*p)  // use text immediately following the option letter, and keep looking
    txt = p;

  while (arg < argc - 1 && *argv[arg + 1] != '-') {
    // use text from subsequent args
    if (!txt.empty()) txt += ' ';  //..space separated if more than one
    txt += argv[++arg];
  }
  Conv.AddOption(ch, typ, txt.c_str());
}

void usage() {
  cout << "Open Babel " << BABEL_VERSION << " -- " << __DATE__ << " -- "
       << __TIME__ << endl;
  cout << "Usage:\n"
       << program_name
       << " [-i<input-type>] <infilename> [-o<output-type>] -O<outfilename> "
          "[Options]"
       << endl;
  cout << "Try  -H option for more information." << endl;

#ifdef _DEBUG
  // CM keep window open
  cout << "Press any key to finish" << endl;
  getch();
#endif
}

void help() {
  cout << "Open Babel converts chemical structures from one file format to "
          "another"
       << endl
       << endl;
  cout << "Usage: " << endl;
  cout << program_name
       << "[-i<input-type>] <infilename> [-o<output-type>] -O<outfilename> "
          "[Options]"
       << endl;
  cout << "The extension of a file decides the format, unless it is overridden"
       << endl;
  cout << " by -i or -o options, e.g. -icml, or -o smi" << endl;
  cout << "See below for available format-types, which are the same as the "
       << endl;
  cout << "file extensions and are case independent." << endl;
  cout
      << "If no input or output file is given stdin or stdout are used instead."
      << endl
      << endl;
  cout
      << "More than one input file can be specified and their names can contain"
      << endl;
  cout << "wildcard chars (* and ?). The format of each file can be different "
          "unless"
       << endl;
  cout << "the -i option has been used, when they are all the same." << endl;
  cout << "By default, the molecules are aggregated in the output file,"
       << endl;
  cout << " but see -m option, Splitting, below.\n" << endl;

  cout << "Options, other than -i -o -O -m, must come after the input files.\n"
       << endl;
  cout << OBConversion::Description();  // Conversion options
  cout << "-H Outputs this help text" << endl;
  cout << "-Hxxx (xxx is file format ID e.g. -Hcml) gives format info" << endl;
  cout << "-Hall Outputs details of all formats" << endl;
  cout << "-V Outputs version number" << endl;
  cout << "-L <category> Lists plugin classes of this category, e.g. <formats>"
       << endl;
  cout << "   Use just -L for a list of plugin categories." << endl;
  cout << "   Use -L <ID> e.g. -L sdf for details of a format or other plugin."
       << endl;
  cout << "-m Produces multiple output files, to allow:" << endl;
  cout << "    Splitting: e.g.        " << program_name
       << " infile.mol -O new.smi -m" << endl;
  cout << "      puts each molecule into new1.smi new2.smi etc" << endl;
  cout << "    Batch conversion: e.g. " << program_name << " *.mol -osmi -m"
       << endl;
  cout << "      converts each input file to a .smi file" << endl;
#ifdef _WIN32
  cout << "   In Windows these can also be done using the forms" << endl;
  cout << "     " << program_name << " infile.mol -O new*.smi and "
       << program_name << " *.mol -O *.smi respectively.\n"
       << endl;
#endif

  OBFormat* pDefault = OBConversion::GetDefaultFormat();
  if (pDefault)
    // some more options probably for OBMol
    cout << pDefault->TargetClassDescription();

  OBFormat* pAPI = OBConversion::FindFormat("obapi");
  if (pAPI) cout << pAPI->Description();

  cout << "To see a list of recognized file formats use\n"
          "  obabel -L formats [read] [write]\n"
       << "To see details and specific options for a particular format, e.g "
          "CML, use\n  obabel -L cml\n"
       << endl;
  // cout << "The following file formats are recognized:" << endl;
  // OBPlugin::List("formats");
  // cout << "\nSee further specific info and options using -H<format-type>, "
  // "e.g. -Hcml" << endl;
}

/* OpenBabel man page*/
/** \page babel a converter for chemistry and molecular modeling data files
*
* \n
* \par SYNOPSIS
*
* \b babel [-H<help-options>] [-V] [-m] [-d] [-h] [-p] [-s<SMARTS-pattern>] [-v<SMARTS-pattern>] [-f<#> -l<#>] [-c] [-x<format-options>] [-i<input-type>] \<infile\> [-o<output-type>] -O\<outfile\>
*
* \par DESCRIPTION
*
* Open Babel is a program designed to interconvert a number of
* file formats currently used in molecular modeling software. \n\n
*
* Note that Open Babel can also be used as a library to interconvert
* many file formats and to provide standard chemistry software routines.
* See the Open Babel web pages (http://openbabel.org) for more
* information.
*
* \par OPTIONS
*
* If only input and output files are given, Open Babel will guess
* the file type from the filename extension. \n\n
*
* \b -V :
*     Output version number and exit \n\n
* \b -H :
*     Output usage information \n\n
* \b -H\<format-ID\> :
*     Output formatting information and options for format specified\n\n
* \b -Hall :
*     Output formatting information and options for all formats\n\n
* \b -i :
*     Specifies input format, see below for the available formats \n\n
* \b -o :
*     Specifies output format, see below for the available formats \n\n
* \b -m :
*     Produce multiple output files, to allow:\n
*      * Splitting one input file - put each molecule into consecutively numbered output files \n
*      * Batch conversion - convert each of multiple input files into a specified output format \n
*     See examples below \n\n
* \b -d :
*     Delete Hydrogens \n\n
* \b -h :
*     Add Hydrogens \n\n
* \b -p :
*     Add Hydrogens appropriate for pH (use transforms in phmodel.txt)  \n\n
* \b -t :
*     All input files describe a single molecule \n\n
* \b -f\<#\> :
*     For multiple entries input, start import at molecule # \n\n
* \b -l\<#\> :
*     For multiple entries input, stop import at molecule # \n\n
* \b -c :
*     Center atomic coordinates at (0,0,0) \n\n
* \b -s\<SMARTS\> :
*     Convert only molecules matching the SMARTS pattern specified \n\n
* \b -v\<SMARTS\> :
*     Convert only molecules \b NOT matching SMARTS pattern specified \n\n
*
* \par FILE FORMATS
*
* The following formats are currently supported by Open Babel:
*  \n    alc -- Alchemy format
*  \n    bgf -- BGF format
*  \n    box -- Dock 3.5 Box format
*  \n    bs -- Ball and Stick format
*  \n    c3d1 -- Chem3D Cartesian 1 format
*  \n    c3d2 -- Chem3D Cartesian2 format
*  \n    caccrt -- Cacao format
*  \n    cache -- CAChe format [Writeonly]
*  \n    cacint -- CacaoInternal format [Writeonly]
*  \n    car -- MSI Biosym/Insight II CAR format [Readonly]
*  \n    ccc -- CCC format [Readonly]
*  \n    cht -- ChemTool format [Writeonly]
*  \n    cml -- Chemical Markup Language
*  \n    com -- Gaussian Cartesian Input [Writeonly]
*  \n    crk2d -- Chemical Resource Kit diagram format (2D)
*  \n    crk3d -- Chemical Resource Kit 3D format
*  \n    csr -- CSR format [Writeonly]
*  \n    cssr -- CSD CSSR format [Writeonly]
*  \n    ct -- ChemDraw Connection Table format
*  \n    dmol -- DMol3 coordinates format
*  \n    ent -- Protein Data Bank format
*  \n    feat -- Feature format
*  \n    fh -- Fenske-Hall Z-Matrix format [Writeonly]
*  \n    fix -- FIX format [Writeonly]
*  \n    g03 -- Gaussian 98/03 Output [Readonly]
*  \n    g98 -- Gaussian 98/03 Output [Readonly]
*  \n    gam -- GAMESS Output
*  \n    gamout -- GAMESS Output
*  \n    gau -- Gaussian Cartesian Input [Writeonly]
*  \n    gjc -- Gaussian Cartesian Input [Writeonly]
*  \n    gjf -- Gaussian Cartesian Input [Writeonly]
*  \n    gpr -- Ghemical format
*  \n    gr96 -- GROMOS96 format [Writeonly]
*  \n    gzmat -- Gaussian Z-Matrix Input
*  \n    hin -- HyperChem Input format
*  \n    ins -- ShelX format [Readonly]
*  \n    jout -- Jaguar output format
*  \n    mdl -- MDL MOL format
*  \n    mm1gp -- Ghemical format
*  \n    mm3 -- MM3 format [Writeonly]
*  \n    mmd -- MacroMod format
*  \n    mmod -- MacroMod format
*  \n    mol -- MDL MOL format
*  \n    mol2 -- Sybyl Mol2 format
*  \n    mopcrt -- MOPAC Cartesian format
*  \n    mopout -- MOPAC Output format [Readonly]
*  \n    mpqc -- MPQC format [Readonly]
*  \n    nwo -- NWChem format
*  \n    pdb -- Protein Data Bank format
*  \n    pov -- POV-Ray input format [Writeonly]
*  \n    pqs -- Parallel Quantum Solutions format
*  \n    prep -- Amber Prep format [Readonly]
*  \n    qcout -- QChem output format
*  \n    qm1gp -- Ghemical format
*  \n    report -- Report format [Writeonly]
*  \n    res -- ShelX format [Readonly]
*  \n    rxn -- MDL RXN format
*  \n    sd -- MDL MOL format
*  \n    sdf -- MDL MOL format
*  \n    smi -- SMILES format
*  \n    tmol -- TurboMole Coordinate format
*  \n    txyz -- Tinker format [Writeonly]
*  \n    unixyz -- UniChem XYZ format
*  \n    vmol -- ViewMol format
*  \n    xed -- XED format [Writeonly]
*  \n    xyz -- XYZ cartesian coordinates format
*  \n    zin -- ZINDO input format [Writeonly]
*
* \par FORMAT OPTIONS
*  Individual file formats may have additional formatting options. \n
*  Input format options are preceded by 'a', e.g. -as \n
*  Output format options are preceded by 'x', e.g. -xn \n
*    For further specific information and options, use -H<format-type> \n
*    e.g., -Hcml
*
* \par EXAMPLES
*  - Standard conversion \n
*     babel -ixyz ethanol.xyz -opdb ethanol.pdb \n
*  - Conversion from a SMI file in STDIN to a Mol2 file written to STDOUT \n
*     babel -ismi -omol2 \n
*  - Split a multi-molecule file into new1.smi, new2.smi, etc. \n
*     babel infile.mol new.smi -m \n
*
* \par AUTHORS
*
* Open Babel is currently maintained by \b Geoff \b Hutchison, \b Chris \b Morley and \b Michael \b Banck.
*
* For more contributors to Open Babel, see http://openbabel.org/THANKS.shtml
*
* \par COPYRIGHT
*  Copyright (C) 1998-2001 by OpenEye Scientific Software, Inc.
*  Some portions Copyright (C) 2001-2005 by Geoffrey R. Hutchison \n \n
*  This program is free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation version 2 of the License.\n \n
*  This program is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
* \par SEE ALSO
*   The web pages for Open Babel can be found at http://openbabel.org/
**/
